schema/messaging.proto

```diff
@@ -0,0 +1,508 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+import "google/api/annotations.proto";
+import "google/longrunning/operations.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/timestamp.proto";
+import "google/rpc/error_details.proto";
+
+package google.showcase.v1alpha3;
+
+option go_package = "github.com/googleapis/gapic-showcase/server/genproto";
+option java_package = "com.google.showcase.v1alpha3";
+option java_multiple_files = true;
+
+option (google.api.metadata) = {
+  product_uri: "https://github.com/googleapis/gapic-showcase"
+};
+
+// This resource is defined in google/showcase/v1alpha3/user.proto. Since
+// google/showcase/v1alpha3/user.proto is not imported from this file, the
+// resource needs to be defined here to be used in this API.
+option (google.api.resource_definition) = {
+  name: "User",
+  path: "users/{user_id}"
+};
+
+// A resource set defining the possible parent paths of a
+// message.
+option (google.api.resource_set_definition) = {
+  name: "MessageParent"
+  resources: [{
+    name: "Profile",
+    path: "users/{user_id}/profile"
+  }]
+  resource_references: [
+    "google.showcase.v1alpha3.Room"
+  ]
+};
+
+// A simple messaging service that implements chat rooms and profile posts.
+//
+// This messaging service showcases the features that API clients
+// generated by gapic-generators implement.
+service Messaging {
+  // Creates a room. The fields "room.name", "room.create_time", and
+  // "room.update_time" will be ignored if passed to this method.
+  rpc CreateRoom(CreateRoomRequest) returns (Room) {
+    option (google.api.http) = {
+      post: "/v1alpha3/rooms"
+      body: "*"
+    };
+
+    // This option signifies that the resulting generated client could include
```

Comment:
Thinking out loud; should we use the `(--` syntax here to distinguish from what you would actually want to end up in user documentation?

(No strong opinion.)

Reply:
Removed per comment below.